### ISSUE TITLE
[CI] 10_parent_child/Parent/child inner hits fails during mixedClusterTest on 6.x

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#27229]
* Build failure link:  https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/577/console
* Error message: 
```
09:44:44   1> [2017-11-02T02:44:09,686][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] Stash dump on failure [{
09:44:44   1>   "stash" : {
09:44:44   1>     "body" : {
09:44:44   1>       "error" : {
09:44:44   1>         "root_cause" : [
09:44:44   1>           {
09:44:44   1>             "type" : "remote_transport_exception",
09:44:44   1>             "reason" : "[node-0][127.0.0.1:33233][indices:admin/create]",
09:44:44   1>             "stack_trace" : "[[node-0][127.0.0.1:33233][indices:admin/create]]; nested: RemoteTransportException[[node-0][127.0.0.1:33233][indices:admin/create]]; nested: IllegalArgumentException[Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]];
09:44:44   1> 	at org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:618)
09:44:44   1> 	at org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:563)
09:44:44   1> 	at org.elasticsearch.rest.BytesRestResponse.build(BytesRestResponse.java:138)
09:44:44   1> 	at org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:96)
09:44:44   1> 	at org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:91)
09:44:44   1> 	at org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:58)
09:44:44   1> 	at org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:91)
09:44:44   1> 	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$3.handleException(TransportMasterNodeAction.java:185)
09:44:44   1> 	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1056)
09:44:44   1> 	at org.elasticsearch.transport.TcpTransport.lambda$handleException$23(TcpTransport.java:1462)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.EsExecutors$1.execute(EsExecutors.java:135)
09:44:44   1> 	at org.elasticsearch.transport.TcpTransport.handleException(TcpTransport.java:1460)
09:44:44   1> 	at org.elasticsearch.transport.TcpTransport.handlerResponseError(TcpTransport.java:1452)
09:44:44   1> 	at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1382)
09:44:44   1> 	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:60)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
09:44:44   1> 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
09:44:44   1> 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:297)
09:44:44   1> 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:413)
09:44:44   1> 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
09:44:44   1> 	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
09:44:44   1> 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
09:44:44   1> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
09:44:44   1> 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
09:44:44   1> 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:134)
09:44:44   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
09:44:44   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:544)
09:44:44   1> 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
09:44:44   1> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
09:44:44   1> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
09:44:44   1> 	at java.lang.Thread.run(Thread.java:748)
09:44:44   1> Caused by: RemoteTransportException[[node-0][127.0.0.1:33233][indices:admin/create]]; nested: IllegalArgumentException[Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]];
09:44:44   1> Caused by: java.lang.IllegalArgumentException: Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:494)
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:350)
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:274)
09:44:44   1> 	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$1.execute(MetaDataCreateIndexService.java:410)
09:44:44   1> 	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:640)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:270)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:195)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:130)
09:44:44   1> 	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
09:44:44   1> 	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210)
09:44:44   1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
09:44:44   1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
09:44:44   1> 	at java.lang.Thread.run(Thread.java:748)
09:44:44   1> "
09:44:44   1>           }
09:44:44   1>         ],
09:44:44   1>         "type" : "illegal_argument_exception",
09:44:44   1>         "reason" : "Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]",
09:44:44   1>         "stack_trace" : "java.lang.IllegalArgumentException: Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:494)
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:350)
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:274)
09:44:44   1> 	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$1.execute(MetaDataCreateIndexService.java:410)
09:44:44   1> 	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:640)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:270)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:195)
09:44:44   1> 	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:130)
09:44:44   1> 	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
09:44:44   1> 	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247)
09:44:44   1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210)
09:44:44   1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
09:44:44   1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
09:44:44   1> 	at java.lang.Thread.run(Thread.java:748)
09:44:44   1> "
09:44:44   1>       },
09:44:44   1>       "status" : 400
09:44:44   1>     }
09:44:44   1>   }
09:44:44   1> }]
09:44:44 FAILURE 0.02s | MixedClusterClientYamlTestSuiteIT.test {p0=/10_parent_child/Parent/child inner hits} <<< FAILURES!
09:44:44    > Throwable #1: java.lang.AssertionError: Failure at [/10_parent_child:2]: expected [2xx] status code but api [indices.create] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"remote_transport_exception","reason":"[node-0][127.0.0.1:33233][indices:admin/create]","stack_trace":"[[node-0][127.0.0.1:33233][indices:admin/create]]; nested: RemoteTransportException[[node-0][127.0.0.1:33233][indices:admin/create]]; nested: IllegalArgumentException[Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]];\n\tat
```

* Reproduction line:
```
gradle :qa:mixed-cluster:v6.0.0-SNAPSHOT#mixedClusterTestRunner \
  -Dtests.seed=EC8A38525928113 \
  -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT \
  -Dtests.method="test {p0=/10_parent_child/Parent/child inner hits}" \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-LY \
  -Dtests.timezone=America/Dawson_Creek
```

I think the interesting part is the rejected Mapping:

```
09:44:44   1>         "reason" : "Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]",
09:44:44   1>         "stack_trace" : "java.lang.IllegalArgumentException: Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]
09:44:44   1> 	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:494)
```

**Comment by cbuescher** [comment]
Another one, in case you need more data points: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/578/console

**Comment by cbuescher** [comment]
Another one here: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/581/console

**Comment by martijnvg** [comment]
Looking at it now and it looks like this test should have been changed to use the join field instead as multiple types can't be created on new indices from 6.0.

**Comment by martijnvg** [comment]
ah I think this should be fixed by only letting this test run against a 5.6.x cluster, right now it runs against a 6.0 cluster too.

**Comment by davidkyle** [comment]
Another instance: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/585/console

This reproduces
```
gradle :qa:mixed-cluster:v6.0.0-SNAPSHOT#mixedClusterTestRunner \
  -Dtests.seed=5C6B7A705BE22340 \
  -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT \
  -Dtests.method="test {p0=/10_parent_child/Parent/child inner hits}" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-IE \
  -Dtests.timezone=US/Samoa
```

```
  1> Caused by: java.lang.IllegalArgumentException: Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]
  1>    at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:494)
  1>    at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:350)
  1>    at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:274)
  1>    at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$1.execute(MetaDataCreateIndexService.java:410)
  1>    at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
  1>    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:640)
  1>    at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:270)
  1>    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:195)
```

**Comment by davidkyle** [comment]
And another one:

```
gradle :qa:mixed-cluster:v6.0.0-SNAPSHOT#mixedClusterTestRunner \
  -Dtests.seed=B24673332795A968 \
  -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT \
  -Dtests.method="test {p0=/10_parent_child/Parent/child inner hits}" \
  -Dtests.security.manager=true \
  -Dtests.locale=de-AT \
  -Dtests.timezone=Asia/Macao
```

With the same error 

```
java.lang.IllegalArgumentException: Rejecting mapping update to [test] as the final mapping would have more than 1 type: [type_3, type_2]
```

**Comment by martijnvg** [comment]
Fixed via: 07b42554843e859d63250a2ac6619232fd61631c
